src/main/java/mat/client/measure/AbstractNewMeasureView.java

```diff
@@ -303,6 +322,26 @@ protected void addCompositeMeasureModelType() {
         measureModelGroup.add(buildCompositeModelTypePanel());
     }
 
+    protected void addGenerateCmsIdCheckbox() {
+        generateCmsIdCheckbox.setText("Automatically generate a CMS ID and matching Library Name upon save");
+        generateCmsIdCheckbox.setTitle("Click to generate a CMS ID and matching Library Name upon save");
```

Comment:
The "upon save" part reads oddly to me, since we're generating the IDs in real-time and not when the user clicks save.

Reply:
https://jira.cms.gov/browse/MAT-2142
Like in the story, Do you want me to add "upon save" in parenthesis? Honestly, I missed it.

Reply:
I changed it to -> Automatically generate an eCQM ID and matching Library Name (upon save)
as suggested by Thomas.

Reply:
That still misrepresents how the ID is generated in real time, not when saved. Hence the edge cases where, on save, there may be conflicts.

If it was generated on save, that would reduce the chance of a race condition with another user, but the user would not have the opportunity to modify the generated values.